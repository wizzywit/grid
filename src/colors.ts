export const red = '#ee312a'
export const blue = '#1600ff'
export const green = '#008001'
export const yellow = '#feff00'
export const cyan = '#B3CBFB'
export const grey = '#999999'