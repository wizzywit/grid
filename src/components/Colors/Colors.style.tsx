import styled from "styled-components";

export const Styles = {
  ColorsContainer: styled.div`
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    /* justify-content: space-between;
    width: 50%; */
    gap: 25px;
    margin-bottom: 50px;
  `,
};
