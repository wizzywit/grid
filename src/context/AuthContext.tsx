import React from "react";

interface DataArray {
  color: string;
  shape: string;
}

export type Filter = (
  c: Array<string> | ((c: Array<string>) => Array<string>)
) => void;

type PerformFilterType = (
  shape: string,
  shapesFilter: Array<string> | undefined,
  filter: Filter | undefined,
  type: string
) => void;

type AuthContextType = {
  login?: () => void;
  logout?: () => void;
  data?: Array<DataArray>;
  shapesFilter?: Array<string>;
  setShapesFilter?: Filter;
  colorsFilter?: Array<string>;
  setColorsFilter?: Filter;
  isLoggedIn?: boolean;
  performSelect?: PerformFilterType;
};
const AuthContext = React.createContext<AuthContextType>({});

export default AuthContext;
