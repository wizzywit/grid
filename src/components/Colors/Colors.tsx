import React, { useContext } from "react";
import { Styles } from "./Colors.style";
import ColorButton from "../ColorButton/ColorButton";
import { red, blue, green, yellow, cyan, grey } from "../../colors";
import AuthContext from "../../context/AuthContext";
import { getColorName } from "../../helper";

const colors = [red, blue, green, yellow, cyan, grey];
const Colors = () => {
  const { colorsFilter } = useContext(AuthContext);
  return (
    <Styles.ColorsContainer>
      {colors.map((color) => (
        <ColorButton
          key={color}
          color={color}
          isActive={colorsFilter?.includes(getColorName(color))}
        />
      ))}
    </Styles.ColorsContainer>
  );
};

export default Colors;
