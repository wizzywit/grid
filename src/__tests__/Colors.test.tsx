import React from "react";

import { render } from "@testing-library/react";

import Colors from "../components/Colors/Colors";
import AuthContext from "../context/AuthContext";
import { red, blue, green, yellow, cyan, grey } from "../colors";

const colors = [red, blue, green, yellow, cyan, grey];

describe("Color", () => {
  test("should display list of colors", async () => {
    const { container, findAllByTestId } = render(
      <AuthContext.Provider value={{ colors }}>
        <Colors />
      </AuthContext.Provider>
    );
    expect(container).toMatchSnapshot();
    const children = await findAllByTestId("color");
    expect(children).toHaveLength(colors.length);
  });
});
