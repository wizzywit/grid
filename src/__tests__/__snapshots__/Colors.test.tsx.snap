// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color should display list of colors 1`] = `
<div>
  <div
    class="sc-bdnxRM gYskiT"
  >
    <button
      class="sc-gtsrHT hEtweE"
      color="#ee312a"
      data-testid="color"
    />
    <button
      class="sc-gtsrHT gAgxHu"
      color="#1600ff"
      data-testid="color"
    />
    <button
      class="sc-gtsrHT eNrMSg"
      color="#008001"
      data-testid="color"
    />
    <button
      class="sc-gtsrHT eORXRq"
      color="#feff00"
      data-testid="color"
    />
    <button
      class="sc-gtsrHT keuowZ"
      color="#B3CBFB"
      data-testid="color"
    />
    <button
      class="sc-gtsrHT iMWynB"
      color="#999999"
      data-testid="color"
    />
  </div>
</div>
`;
